### PR TITLE
Fix #2689: Corrected options in exploration with id text_exp_id_4 of Second Test Topic.

### LINKED ISSUE #2689: Not able to submit answer in Second Test Topic.
**Describe the bug**
There is one option missing in the exploration` test_exp_id_4` of `test_topic_id_1`
**To Reproduce**
1. Go to Second Test Topic on HomeScreen.
2. Click on First chapter 
3. Re-arrange the correct options and submit the answer

**Expected behavior**
There is a need to add all the correct options in `test_exp_id_4.json` file in `IsEqualToOrdering` field which accepts the correct ordering of options.

**Screenshots**
Screenshot_20210211_215153

**Device**
- Vivo v3 
- API level - 21

**Additional context**
This fix has to be done in `StateRetrieverTest` too.

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation
This PR fixes #2689. This includes change in Json data of `text_exp_id_4`, now the file contains options, that has correct ordering of answers. This also fixes the `StateRetrieverTest`

**Screenshot**
Screenshot_20210212_003932

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/android/app/home/topiclist/PromotedItemViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package org.oppia.android.app.home.topiclist
+
+import org.oppia.android.app.viewmodel.ObservableViewModel
+
+/** The root [ViewModel] for all individual items that may be displayed in home fragment recycler view. */
+abstract class PromotedItemViewModel : ObservableViewModel()
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/home/topiclist/PromotedItemViewModel.kt]: Why is this file needed in this PR?

File: app/src/main/res/drawable/bottom_left_rounded_rect_coming_soon_background.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  <corners android:bottomRightRadius="0dp"
+    android:bottomLeftRadius="12dp"
+    android:topLeftRadius="0dp"
+    android:topRightRadius="4dp"/>
+  <gradient
+    android:angle="360"
+    android:startColor="#7659B6"
+    android:endColor="#674172"
+    android:type="linear" />
+</shape>
```
Comment (rt4914) [inline app/src/main/res/drawable/bottom_left_rounded_rect_coming_soon_background.xml]: Ditto for this file, why is this file needed in this PR?

File: app/src/main/res/layout/promoted_story_list.xml [hunk 3]
```diff
@@ -12,6 +12,7 @@
   </data>
 
   <LinearLayout
+    android:id="@+id/promoted_story_list_container"
```
Comment (rt4914) [inline app/src/main/res/layout/promoted_story_list.xml]: Ditto

File: app/src/main/res/layout/fraction_interaction_item.xml [hunk 4]
```diff
@@ -28,11 +28,11 @@
       android:layout_width="match_parent"
       android:layout_height="wrap_content"
       android:background="@drawable/edit_text_background"
-      android:digits="0123456789/- "
+      android:digits="0123456789/-"
       android:fontFamily="sans-serif"
       android:hint="@{viewModel.hintText}"
       android:imeOptions="actionDone"
-      android:inputType="date"
+      android:inputType="text"
```
Comment (rt4914) [inline app/src/main/res/layout/fraction_interaction_item.xml]: Ditto

File: domain/src/main/assets/coming_soon_topics.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "topic_id_list": [
+    "test_topic_id_0",
+    "test_topic_id_1",
+    "GJ2rLXRKD5hw",
+    "omzF4oqgeTXd"
+  ]
+}
```
Comment (rt4914) [inline domain/src/main/assets/coming_soon_topics.json]: Ditto

**Review by rt4914 — CHANGES_REQUESTED** [review]
This solution works but the PR introduces a lot of other changes which should not be part of this PR.

File: app/src/main/java/org/oppia/android/app/home/topiclist/PromotedItemViewModel.kt (hunk 1 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/android/app/home/topiclist/PromotedItemViewModel.kt]: Fixed.

File: app/src/main/res/drawable/bottom_left_rounded_rect_coming_soon_background.xml (hunk 2 above)
Comment (veena14cs) [reply app/src/main/res/drawable/bottom_left_rounded_rect_coming_soon_background.xml]: Fixed.

File: app/src/main/res/layout/promoted_story_list.xml (hunk 3 above)
Comment (veena14cs) [reply app/src/main/res/layout/promoted_story_list.xml]: Fixed.

File: app/src/main/res/layout/fraction_interaction_item.xml (hunk 4 above)
Comment (veena14cs) [reply app/src/main/res/layout/fraction_interaction_item.xml]: Fixed.

File: domain/src/main/assets/coming_soon_topics.json (hunk 5 above)
Comment (veena14cs) [reply domain/src/main/assets/coming_soon_topics.json]: Fixed.

**Comment by veena14cs** [comment]
@rt4914 fixed unrelated file changes in the PR PTAL.

**Review by BenHenning — APPROVED** [review]
Thanks @veena14cs.

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks

**Comment by rt4914** [comment]
Checks are failing. Rerunning the now.

**Pull request merged**